agent/games/game_engine.py

```
@@ -0,0 +1,429 @@
+        
+        if client_pattern != expected_pattern:
+            return False, "Pattern mismatch"
+        
+        if client_time > max_time:
+            return False, f"Time exceeded: {client_time}ms (max {max_time}ms)"
+        
+        return True, None
+    
+    def _validate_dice_roll(
+        self,
+        challenge: GameChallenge,
```

Ruff (0.14.10)

Unused method argument: `challenge`

(ARG002)